src/pywws/service/pgsql.py

```diff
@@ -0,0 +1,277 @@
+import psycopg2.extras
+import time as t
+import math
+from pywws.conversions import dew_point
+
+
+__docformat__ = "restructuredtext en"
+service_name = os.path.splitext(os.path.basename(__file__))[0]
+logger = logging.getLogger(__name__)
+RTFREQ = timedelta(seconds=48)
+
+
+class DB():
+    """Class for database handling"""
+    def __init__(self, connstr):
+        """Try connecting to PGSQL"""
+        try:
+            self.conn = psycopg2.connect(connstr)
+            self.c = self.conn.cursor(cursor_factory=psycopg2.extras.DictCursor)
+        except:
+            print 'ERROR:  Unable to connect to PostgreSQL'
+
+    def query(self, query):
+        """Send a query and return result"""
+        self.c.execute(query)
+        self.conn.commit()
+        try:
+            return self.c.fetchall()
+        except psycopg2.ProgrammingError:
+            return None
+
+    def queryone(self, query):
+        """Send query and expect one row"""
+        self.c.execute(query)
+        self.conn.commit()
+        try:
+            return self.c.fetchone()
+        except psycopg2.ProgrammingError:
+            return None
+
+    def close(self):
+        """Close connection to PGSQL"""
+        self.conn.close()
+
+    def __delete__(self):
+        """Deconstructor"""
+        self.close()
+
+
+class ToService(pywws.service.CatchupDataService):
+    config = {
+        'host': ('', True, 'HOST'),
+        'username': ('', True, 'USERNAME'),
+        'password': ('', True, 'PASSWORD'),
+        'database': ('', True, 'DATABASE NAME'),
+        'table': ('', True, 'DATABASE TABLE'),
+        }
+    fixed_data = {'action': 'updateraw', 'softwaretype': 'pywws'}
+    interval = timedelta(seconds=47)
+    logger = logger
+    service_name = service_name
+    template = """
+#live#
+#idx          "'dateutc'     : '%Y-%m-%d %H:%M:%S',"#
+#wind_dir     "'winddir'     : '%.0f'," "" "winddir_degrees(x)"#
+#wind_ave     "'windspeedmph': '%.2f'," "" "wind_mph(x)"#
+#wind_gust    "'windgustmph' : '%.2f'," "" "wind_mph(x)"#
+#hum_out      "'humidity'    : '%.d',"#
+#temp_out     "'tempf'       : '%.1f'," "" "temp_f(x)"#
+#rel_pressure "'baromin'     : '%.4f'," "" "pressure_inhg(x)"#
+#calc "temp_f(dew_point(data['temp_out'], data['hum_out']))" "'dewptf': '%.1f',"#
+#calc "rain_inch(rain_hour(data))" "'rainin': '%g',"#
+#calc "rain_inch(rain_day(data))" "'dailyrainin': '%g',"#
+#hum_in       "'indoorhumidity': '%.d',"#
+#temp_in      "'indoortempf'   : '%.1f'," "" "temp_f(x)"#
+"""
+
+    def __init__(self, context, check_params=True):
+        super(ToService, self).__init__(context, check_params)
+        self.wstype = 1080
+        # extend template
+        if context.params.get('config', 'ws type') == '3080':
+            self.wstype = 3080
+            self.template += """
+#illuminance  "'solarradiation': '%.2f'," "" "illuminance_wm2(x)"#
+#uv           "'UV'            : '%d',"#
+"""
+
+    @contextmanager
+    def session(self):
+        yield None
+
```

Comment:
The raw data is already in Celsius. Instead of using this you could simply remove the Fahrenheit conversions from your template.